internals/vector.md

```diff
@@ -0,0 +1,118 @@
+---
+layout: docu
+title: Execution Format
+---
+
+`Vector` is the container format used to store in-memory data during execution.  
+`DataChunk` is a collection of Vectors, used for instance to represent a column list in a PhysicalProjection operator.
+
+### Data Flow
+
+DuckDB uses a vectorized query execution model. This means that our operators are optimized to work on Vectors of a fixed size.
+This fixed size is commonly referred to in the code as `STANDARD_VECTOR_SIZE`.  
+The default STANDARD_VECTOR_SIZE is 2048 tuples.
+
+### Internals
+
+`ValidityMask validity`  
+Tracks which elements in the Vector are NULL.  
+
+`data_ptr_t data`  
+The element data of the Vector.  
+
+`VectorBuffer buffer`  
+An interface implemented based on the type of the vector and the data type it's representing.  
+Usually owns the `data` memory.  
+
+`VectorBuffer auxiliary`  
+Optionally stores additional data needed by the Vector, explained later.  
+
+
+### Vector Format
+
+To efficiently represent different shapes of collections, we use different Vector formats.
+
+#### FLAT_VECTOR
+
+Flat vector is a sequential array, this is the default format.
+
+#### CONSTANT_VECTOR
+
+Constant vector is used to represent a single element as an entire vector.  
+For example:  
+```sql
+select lst, 'duckdb' from range(1000) tbl(lst);
+```
+Since `duckdb` is Constant Expression, we know it's a single entry, but operators expect a DataChunk to have the same amount of tuples.  
+If we only had FlatVector we would have to duplicate 'duckdb' a thousand times.  
+Instead we can use a ConstantVector to only store 'duckdb' once.
+
+#### DICTIONARY_VECTOR
+
+Dictionary vector is used when a transformation would not change any elements of the data, but only change the order of the elements.  
+For example:  
+```sql
+select * from range(1000) order by 1 desc;
+```
+Instead of duplicating the result of the `range` result and sorting that, we can use the same data, but just represent it in a different shape.  
```

Comment:
Language nitpick - can we remove all instances of `just` in the text?

Reply:
Yea I agree, re-reading it it sounds sloppy